executor/executor.go

```diff
@@ -21,7 +23,10 @@ func (e ErrStatus) Error() string {
 }
 
 // Run executes a slice of CommandDefs
-func Run(output io.Writer, job screwdriver.JobDef) error {
+func Run(path string, output io.Writer, job screwdriver.JobDef) error {
+	if err := chdir(path); err != nil {
```

Comment:
To address the concern of changing directories from a helper function call,
I believe we can just reset the working directory once all commands have
been executed.

As for propagating working directories, perhaps we can track the directory
at the end of each command execution and set  the dirs of subsequent calls